src/player_activity.cpp

```diff
@@ -33,6 +33,37 @@ player_activity::player_activity( activity_type t, int turns, int Index, int pos
 {
 }
 
+player_activity::player_activity( const player_activity &rhs )
+    : type( rhs.type ), moves_total( rhs.moves_total ), moves_left( rhs.moves_left ),
+      index( rhs.index ), position( rhs.position ), name( rhs.name ),
+      ignore_trivial( rhs.ignore_trivial ), values( rhs.values ), str_values( rhs.str_values ),
+      coords( rhs.coords ), placement( rhs.placement ),
+      warned_of_proximity( rhs.warned_of_proximity ), auto_resume( rhs.auto_resume )
+{
+    target = rhs.target.clone();
```

Comment:
Could probably be moved to the initializer list as `target( rhs.target.clone() )`.

We could also drop the whole no copying constructor thing from item locations. I added it at start when they were supposed not to be saved or copied, but were guaranteed to point to a valid item. Now they are both saved and copied, and the `target()` function guarantees it points to a valid item or `null`, so the original rationale no longer applies.